static/js/src/advantage/react/components/Subscriptions/SubscriptionEdit/SubscriptionEdit.tsx

```diff
@@ -0,0 +1,23 @@
+import { Button } from "@canonical/react-components";
+import React from "react";
+import usePortal from "react-useportal";
+
+import SubscriptionCancel from "../SubscriptionCancel";
+
+const SubscriptionEdit = () => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  return (
+    <>
+      <Button appearance="link" data-test="cancel-button" onClick={openPortal}>
+        You can cancel this subscription online or contact us.
+      </Button>
+      {isOpen && (
+        <Portal>
```

Comment:
What is the Portal used for here? I tried to remove it and saw no difference.
If it's needed, do you think it should be a part of `SubscriptionCancel` or even the React Vanilla modal component?

Reply:
`usePortal` is a wrapper around [React portals](https://reactjs.org/docs/portals.html) which means that the portal gets pushed to the top of the DOM so that it is not constrained by it's current position in the DOM. e.g. for a modal if the parent had `position: relative` or `overflow: hidden` it wouldn't fill the whole screen.